### ISSUE TITLE
Callback video painters probably need to take into account track information for correct aspect ratio

### ISSUE THREAD

**Issue by caprica** [caprica/vlcj#833]
This is mostly apparent in DVD video which currently looks "wrong" when played with the CallbackMediaPlayerComponent.

Example DVD track information:
```
[VideoTrackInfo[codec=0x7667706d,codecName=mpgv,originalCodec=0x7667706d,originalCodecName=mpgv,id=2,profile=-1,level=-1,bitRate=0,language=null,description=null,codecDescription=MPEG-1/2 Video][width=720,height=576,sampleAspectRatio=64,sampleAspectRatioBase=45,frameRate=25,frameRateBase=1,orientation=TOP_LEFT,projection=RECTANGULAR,yaw=0.0,pitch=0.0,roll=0.0,fov=80.0,multiview=null]]
```
DVD video has a video frame buffer dimension of 720x576, i.e. a ratio 1.25.

The track info example has an aspect ratio of 64:45, i.e. 1.4222.

Consequently the DVD video does not scale correctly.

No big deal, right, just get the SAR and use it?

However, the track information does not appear until some short time after the video has started playing, so we are potentially already rendering frames at the wrong size.

Whether this is an easy or very difficult indeed thing to resolve is at what point can we get the video track information.

**Comment by caprica** [comment]
Only partial track information is available in the setup callback, there is no valid SAR.

In fact, the SAR does not get filled even on the first display callback, it was the second. (Presumably it may be later).

So this means some frames will be rendered with the wrong aspect.

**Comment by caprica** [comment]
This is actually a problem.

With DVD, media player is telling me current track is 224 (that's fine, they don't have to start from zero).

But I can't get the video track info for track 224, it has an id of 1.

There is no way to link the two things together reliably (at least for DVD).

**Comment by caprica** [comment]
Punting into the long grass for now.

In any case, as stated above, there's no reliable way to determine the SAR for the current track.

I would like to revisit this one, but right now I don't see a way forward.

**Comment by caprica** [comment]
Resolved by #986, made possible by the new native track selection API with LibVLC 4.0.0.
